ui/pages/notifications/notification-components/snap/snap.tsx

```diff
@@ -6,68 +6,93 @@ import {
-  const dispatch = useDispatch();
-  const history = useHistory();
-  const trackEvent = useContext(MetaMetricsContext);
-
-  const snapsMetadata = useSelector(getSnapsMetadata);
-
-  const snapsNameGetter = getSnapName(snapsMetadata);
-
-  const handleSnapClick = () => {
-    dispatch(markNotificationsAsRead([snapNotification.id]));
-    trackEvent({
-      category: MetaMetricsEventCategory.NotificationInteraction,
-      event: MetaMetricsEventName.NotificationClicked,
-      properties: {
-        notification_id: snapNotification.id,
-        notification_type: snapNotification.type,
-        previously_read: snapNotification.isRead,
-      },
-    });
-  };
-
-  const handleSnapButton = () => {
-    dispatch(markNotificationsAsRead([snapNotification.id]));
-    trackEvent({
-      category: MetaMetricsEventCategory.NotificationInteraction,
-      event: MetaMetricsEventName.NotificationClicked,
-      properties: {
-        notification_id: snapNotification.id,
-        notification_type: snapNotification.type,
-        previously_read: snapNotification.isRead,
-      },
-    });
-    history.push(getSnapRoute(snapNotification.data.origin));
-  };
-
-  return (
-    <NotificationListItemSnap
-      id={snapNotification.id}
-      snapId={snapNotification.data.origin}
-      isRead={snapNotification.isRead}
-      createdAt={new Date(snapNotification.createdAt)}
-      title={{
-        items: [
-          {
-            text: snapsNameGetter(snapNotification.data.origin) || 'Snap',
-          },
-        ],
-      }}
-      snapMessage={snapNotification.data.message}
-      handleSnapClick={handleSnapClick}
-      handleSnapButton={handleSnapButton}
-    />
-  );
+export const components: NotificationComponent<SnapNotification> = {
+  guardFn: isOfTypeNodeGuard(['snap' as typeof TRIGGER_TYPES.SNAP]),
+  item: ({ notification, onClick }) => {
+    const dispatch = useDispatch();
+    const history = useHistory();
+    const trackEvent = useContext(MetaMetricsContext);
+    const snapsMetadata = useSelector(getSnapsMetadata);
+    const snapsNameGetter = getSnapName(snapsMetadata);
+    const handleSnapButton = () => {
+      dispatch(markNotificationsAsRead([notification.id]));
+      trackEvent({
+        category: MetaMetricsEventCategory.NotificationInteraction,
+        event: MetaMetricsEventName.NotificationClicked,
+        properties: {
+          notification_id: notification.id,
+          notification_type: notification.type,
+          previously_read: notification.isRead,
+        },
+      });
+      history.push(getSnapRoute(notification.data.origin));
+    };
+    return (
+      <NotificationListItemSnap
+        id={notification.id}
+        snapId={notification.data.origin}
+        isRead={notification.isRead}
+        createdAt={new Date(notification.createdAt)}
+        title={{
+          items: [
+            {
+              text: snapsNameGetter(notification.data.origin) || 'Snap',
```

Comment:
Maybe use "Unknown Snap"? Is there a case in which this happens in the first place?
```suggestion
              text: snapsNameGetter(notification.data.origin) ?? 'Unknown Snap',
```

Reply:
None that I know of, i think the notifications team just had it as a fail-safe. I'll change it.